config/loader.js

```diff
@@ -0,0 +1,243 @@
+      hasDefault = true;
+      dotenv.config({
+        path: this.resolve(this.envMap.default),
+      });
+    } else {
+      console.warn('The default .env file was not found');
+    }
+    
+    const environment = this.currentEnvironment();
+
+    // Load the environment specific env file
+    const envFile = this.envMap[environment];
+
+    // check if the file exists
+    if (fs.existsSync(envFile)) {
+      dotenv.config({
+        path: this.resolve(envFile),
+      });
+    } else if (!hasDefault) {
+      console.warn('No env files found, have you completed the install process?');
+    }
+  }
+
+  /**
+   * Validate Config
+   */
+  validate() {
+    const requiredKeys = [
+      'NODE_ENV',
+      'JWT_SECRET',
+      'DOMAIN_CLIENT',
+      'DOMAIN_SERVER',
+      'CREDS_KEY',
+      'CREDS_IV',
+    ];
+
+    const missingKeys = requiredKeys.map(key => {
+      const variable = process.env[key];
+      if (variable === undefined || variable === null) {
+        return key;
+      }
+    }).filter(value => value !== undefined);
+
+    // Throw an error if any required keys are missing
+    if (missingKeys.length) {
+      const message = `
+        The following required env variables are missing:
+            ${missingKeys.toString()}.
+        Please add them to your env file or run 'npm run install'
+      `;
+      throw new Error(message);
+    }
+
+    // Check JWT secret for default
+    if (process.env.JWT_SECRET === 'secret') {
+      console.warn('Warning: JWT_SECRET is set to default value');
+    }
+  }
+
+  /**
+   * Resolve the location of the env file
+   *
+   * @param {String} envFile 
+   * @returns 
+   */
+  resolve(envFile) {
+    return path.resolve(process.cwd(), envFile);
+  }
+
+  /**
+   * Add secure keys to the env
+   *
+   * @param {String} filePath The path of the .env you are updating
+   * @param {String} key The env you are adding
+   * @param {Number} length The length of the secure key
+   */
+  addSecureEnvVar(filePath, key, length) {
+    const env = {};
+    env[key] = this.generateSecureRandomString(length);
+    this.writeEnvFile(filePath, env);
+  }
+
+  /**
+   * Write the change to the env file
+   */
+  writeEnvFile(filePath, env) {
+    const content = fs.readFileSync(filePath, 'utf-8');
+    const lines = content.split('\n');
+    const updatedLines = lines.map(line => {
+      if (line.trim().startsWith('#')) {
+        // Allow comment removal
+        if (env[line] === 'remove') {
+          return null; // Mark the line for removal
+        }
+        // Preserve comments
+        return line;
+      }
+
+      const [key, value] = line.split('=');
+      if (key && value && env.hasOwnProperty(key.trim())) {
+        if (env[key.trim()] === 'remove') {
+          return null; // Mark the line for removal
+        }
+        return `${key.trim()}=${env[key.trim()]}`;
+      }
+      return line;
+    }).filter(line => line !== null); // Remove lines marked for removal
```

Comment:
I think there is a bug here with the line splitting, specifically for mongodb. (because there is more than one = sign in the string

Reply:
although the value isn't used, so maybe its okay

Reply:
its fine here but not further down, see other comment